distage/distage-core-api/src/main/scala/izumi/distage/model/definition/DIResource.scala

```diff
@@ -554,15 +554,15 @@ object DIResource {
       evalMap[G, OuterResource](a => DIEffect[G].map(f(a))(_ => a))
 
     /** Wrap acquire action of this resource in another effect, e.g. for logging purposes */
-    final def wrapAcquire[G[x] >: F[x]](f: (=> G[InnerResource]) => G[InnerResource]): DIResourceBase[G, OuterResource] =
-      wrapAcquireImpl[G, OuterResource](this: this.type)(f)
+    final def wrapAcquire[G[x] >: F[x], I2 >: InnerResource](f: (=> G[InnerResource]) => G[I2]): DIResourceBase[G, OuterResource] =
```

Comment:
This is unsound, e.g.  subsequent `release` will throw `ClassCastException` if `I2`=Any

Reply:
@neko-kai Yup, I've reverted these changes. Was trying to deal with DIResource in tree traverse with action wrapping, but it was overcomplicated.